pkg/registry/pod/strategy.go

```diff
@@ -251,8 +266,11 @@ func LogLocation(
 	if len(container) == 0 {
 		if len(pod.Spec.Containers) == 1 {
 			container = pod.Spec.Containers[0].Name
-		} else {
+		} else if len(pod.Spec.Containers) == 0 {
 			return nil, nil, errors.NewBadRequest(fmt.Sprintf("a container name must be specified for pod %s", name))
+		} else {
```

Comment:
Don't use else blocks when you return from a condition early - this would be better as a switch block (we generally don't use if else if)